File: exercise10/SelfCheckout.java
package exercise10;

import java.util.Scanner;

public class SelfCheckout {
    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);
        final double TAX_RATE = 5.5;
        int item1, item2, item3;
        int qty1, qty2, qty3;

        System.out.print("Enter the price of item 1: ");
        item1 = scanner.nextInt();
        System.out.print("Enter the quantity of item 1: ");
        qty1 = scanner.nextInt();

        System.out.print("Enter the price of item 2: ");
        item2 = scanner.nextInt();
        System.out.print("Enter the quantity of item 2: ");
        qty2 = scanner.nextInt();

        System.out.print("Enter the price of item 3: ");
        item3 = scanner.nextInt();
        System.out.print("Enter the quantity of item 3: ");
        qty3 = scanner.nextInt();

        scanner.close();


        double subtotal = (item1 * qty1) + (item2 * qty2) + (item3 * qty3);
        double tax = subtotal * TAX_RATE / 100.0;
        double total = subtotal + tax;


        System.out.printf("Subtotal: €%.2f\n", subtotal);
        System.out.printf("Subtotal: €%.2f\n", tax);
        System.out.printf("Subtotal: €%.2f\n", total);
    }
}

File: 2. Calculations/exercise10/python/self_checkout.py
# Exercise 10 - Self Checkout

# Declare constant
TAX_RATE = 5.5

# Get input from user and convert to relevant type
item1 = input('Enter the price of item 1: ')
qty1 = input('Enter the quantity of item 1: ')

item1 = float(item1)
qty1 = int(qty1)

item2 = input('Enter the price of item 2: ')
qty2 = input('Enter the quantity of item 2: ')

item2 = float(item2)
qty2 = int(qty2)

item3 = input('Enter the price of item 3: ')
qty3 = input('Enter the quantity of item 3: ')

item3 = float(item3)
qty3 = int(qty3)


# Perform calculations
subtotal = (item1 * qty1) + (item2 * qty2) + (item3 * qty3)
tax = subtotal * TAX_RATE / 100.0
total = subtotal + tax


# Generate output text
output_text = f'Subtotal: €{subtotal:.2f}\n' \
              f'Subtotal: €{tax:.2f}\n' \
              f'Subtotal: €{total:.2f}\n'

print(output_text)
